File: 2958-length-of-longest-subarray-with-at-most-k-frequency/2958-length-of-longest-subarray-with-at-most-k-frequency.py
class Solution:
    def maxSubarrayLength(self, nums: List[int], k: int) -> int:
        hash_map = {}
        l = maxLen = 0
        for r in range(len(nums)):
            hash_map[nums[r]] = hash_map.get(nums[r] , 0) + 1
            while hash_map[nums[r]] > k:
                hash_map[nums[l]] -= 1
                l += 1
            
            maxLen = max(maxLen , r - l + 1)
        return maxLen

File: 2958-length-of-longest-subarray-with-at-most-k-frequency/2958-length-of-longest-subarray-with-at-most-k-frequency.cpp
class Solution {
public:
    int maxSubarrayLength(vector<int>& nums, int k) {
        int n=nums.size();
        unordered_map<int,int> mymap;
        int i=0,j=0;
        int ans=0;
        while (j<n){
            mymap[nums[j]]++;
            
            
            while (mymap[nums[j]]>k){
                mymap[nums[i]]--;
                i++;

            }
            ans=max(ans,j-i+1);
            j++;
        }
        return ans;
        
    }
};